clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java

```diff
@@ -110,32 +108,24 @@ CompletionStage<Void> bootstrapFromBlobs(Store store, int versionNumber, int par
     String storeName = store.getName();
     String baseDir = serverConfig.getRocksDBPath();
     try {
-      CompletableFuture<InputStream> p2pFuture =
-          blobTransferManager.get(storeName, versionNumber, partitionId).toCompletableFuture();
-      LOGGER.info(
-          "Bootstrapping from blobs for store {}, version {}, partition {}",
+      blobTransferManager.get(storeName, versionNumber, partitionId).toCompletableFuture();
```

Comment:
I think it is good to move all the peer discovery logic into the method, but there is gotcha here. It looks like you make this method a blocking call, and if you check the upstream caller method - VersionBackend.subscribe(), it is iterating over the list of partitions as previously it assumes 
        backend.getIngestionBackend().startConsumption(config, partition);
is a fast action, but now we make the blob transfer for multiple partitions sequential. 
I think we need to make them run in parallel, otherwise we are not utilizing the HW of DVC host.

Reply:
I see, that's fine if we want to fix it separately (but it might impact your batch only store benchmark though, unless you want to resize the store to only single partition)